web/html/src/manager/virtualization/nets/network-properties.test.js

```diff
@@ -0,0 +1,367 @@
+
+    await renderWithNetwork();
+    await select(screen.getByLabelText(/^Network type/), "nat")
+    await type(screen.getByLabelText(/^Name/), "nat0");
+    await type(screen.getByLabelText(/^Maximum Transmission Unit/), "7000");
+    await type(screen.getByRole("textbox", {name: "IPv4 Network address"}), "192.168.10.0");
+    await type(screen.getByRole("textbox", {name: "IPv4 Network address prefix"}), "24");
+    await type(screen.getByRole("textbox", {name: "NAT IPv4 range start"}), "192.168.10.3");
+    await type(screen.getByRole("textbox", {name: "NAT IPv4 range end"}), "192.168.10.4");
+    await type(screen.getByRole("textbox", {name: "NAT port range start"}), "1234");
+    await type(screen.getByRole("textbox", {name: "NAT port range end"}), "1236");
+    expect(screen.getByText("Submit").disabled).toBeFalsy();
+    click(screen.getByText("Submit"));
+  });
+
+  test("Create network with all addressing fields", async () => {
+    onSubmit = ({definition}) => {
+      expect(definition).toStrictEqual({
+        type: "open",
+        bridge: "virbr2",
+        name: "open0",
+        domain: "tf.local",
+        ipv4: {
+          address: "192.168.10.0",
+          prefix: 24,
+          dhcpranges: [
+            {start: "192.168.10.10", end: "192.168.10.20"},
+            {start: "192.168.10.110", end: "192.168.10.120"},
+          ],
+          hosts: [
+            {mac: "2A:C3:A7:A6:01:00", name: "dev-srv", ip: "192.168.10.2"},
+            {mac: "2A:C3:A7:A6:01:01", ip: "192.168.10.3"},
+          ],
+          bootpfile: "pxelinux.0",
+          bootpserver: "192.168.10.2",
+          tftp: "/path/to/tftproot"
+        },
+        ipv6: {
+          address: "2001:db8:ac10:fd01::",
+          prefix: 64,
+          dhcpranges: [
+            {start: "2001:db8:ac10:fd01::10", end: "2001:db8:ac10:fd01::20"},
+          ],
+          hosts: [
+            {id: "0:3:0:1:0:16:3e:11:22:33", name: "peter.xyz", ip: "2001:db8:ac10:fd01::2"},
+          ],
+        },
+        dns: {
+          hosts: [
+            {address: "192.168.10.1", names: ["host", "gateway"]},
+          ],
+          srvs: [
+            {
+              name: "srv1",
+              protocol: "tcp",
+              domain: 'test-domain-name.com',
+              target: 'test.example.com',
+              port: 1111,
+              priority: 11,
+              weight: 111,
+            },
+            {
+              name: "srv2",
+              protocol: "udp",
+            },
+          ],
+          txts: [{name: "example", value: "foo"}, {name: "bar", value: "other value"}],
+          forwarders: [
+            {address: "8.8.4.4"},
+            {address: "192.168.1.1", domain: "example.com"},
+            {domain: "acme.com"},
+          ],
+        },
+      });
+    };
+
+    await renderWithNetwork();
+    await select(screen.getByLabelText(/^Network type/), "open")
+    fireEvent.change(screen.getByLabelText(/^Name/), {target: {value: "open0"}});
```

Comment:
If this works and the current type implementation doesn't, I would move this logic into `type()` instead and verify that it works everywhere. Since it isn't clear without context why this logic is used here but `type()` elsewhere, it's unlikely someone else would come around and fix this later.

Reply:
on that one I have mixed feelings... indeed it would be better to use `type()` and fix it. It would also avoid depending on the components implementation in the tests, but on the other hand tests miserably fail when using `type()`...